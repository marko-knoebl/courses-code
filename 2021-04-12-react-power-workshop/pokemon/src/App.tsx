/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import { NavLink, Routes, Route } from "react-router-dom";

import "./styles.css";
import HomeView from "./home/HomeView";
import PokemonView from "./pokemon/PokemonView";
import CartView from "./cart/CartView";
import { useCart } from "./useCart";
import { usePokemons } from "./usePokemons";
import CartContext from "./CartContext";
import DocumentTitle from "./components/DocumentTitle";
import CounterThatLogsEverySecond from "./CounterThatLogsEverySecond";

function App() {
  const { cart, addToCart, removeFromCart } = useCart();

  const { loadedPokemons, loadMore, requestPokemon } = usePokemons();

  return (
    <CartContext.Provider
      value={{
        cart: cart,
        addToCart: addToCart,
        removeFromCart: removeFromCart,
      }}
    >
      <DocumentTitle
        value={`Pokémon Poster Shop (${cart.length} items in the cart)`}
      />
      <CounterThatLogsEverySecond />
      <div
        css={css`
          background-color: #f0f0f0;
        `}
      >
        <h1>Pokémon poster shop</h1>
        <NavLink to="/">Home</NavLink> |{" "}
        <NavLink to="/cart">
          {cart.length === 0 ? "Cart" : `Cart (${cart.length})`}
        </NavLink>
        <Routes>
          <Route
            path="/"
            element={<HomeView pokemons={loadedPokemons} loadMore={loadMore} />}
          />
          <Route
            path="/pokemon/:id"
            element={
              <PokemonView
                pokemons={loadedPokemons}
                requestPokemon={requestPokemon}
                addToCart={addToCart}
                onRemoveFromCart={removeFromCart}
                cart={cart}
              />
            }
          />
          <Route
            path="/cart"
            element={<CartView pokemons={loadedPokemons} />}
          />
        </Routes>
      </div>
    </CartContext.Provider>
  );
}

export default App;
