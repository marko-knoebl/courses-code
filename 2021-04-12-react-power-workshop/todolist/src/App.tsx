import { useEffect, useReducer, useState } from "react";
import AddTodo from "./AddTodo";
import AddTodoClass from "./AddTodoClass";
import Rating from "./Rating";
import { Todo } from "./Todo";
import TodoItem from "./TodoItem";
import { fetchTodos } from "./todosApi";
import { NavLink, Routes, Route } from "react-router-dom";
import AboutPage from "./AboutPage";

/*
Wir erstellen eine Todo-Anwendung mit der folgenden Funktionalität:

+ Anzeigen erledigter und nicht-erledigter Todos
+ Hinzufügen eines Todos mittels eines Formulars
+ Umschalten des erledigt-Zustandes eines Todos
- Löschen eines Todos
*/

const initialTodos: Array<Todo> = [
  { id: 1, title: "Einkaufen", completed: false },
  { id: 2, title: "Steuererklärung", completed: true },
  { id: 3, title: "Aufräumen", completed: false },
];

type AddTodoAction = {
  type: "addTodo";
  payload: string;
};
type ChangeCompletedAction = {
  type: "changeCompleted";
  payload: { id: number; completed: boolean };
};
type DeleteTodoAction = {
  type: "deleteTodo";
  payload: number;
};
type LoadTodosAction = {
  type: "loadTodos";
  payload: Array<Todo>;
};
type TodosAction =
  | AddTodoAction
  | ChangeCompletedAction
  | DeleteTodoAction
  | LoadTodosAction;

function todosReducer(
  todosState: Array<Todo>,
  action: TodosAction
): Array<Todo> {
  switch (action.type) {
    case "addTodo":
      const newId = Math.max(...todosState.map((t) => t.id), 0) + 1;
      return [
        ...todosState,
        { id: newId, title: action.payload, completed: false },
      ];
    case "deleteTodo":
      return todosState.filter((todo) => todo.id !== action.payload);
    case "changeCompleted":
      return todosState.map((todo) =>
        todo.id === action.payload.id
          ? { ...todo, completed: action.payload.completed }
          : todo
      );
    case "loadTodos":
      return action.payload;
    default:
      return todosState;
  }
}

function App() {
  const [rating1, setRating1] = useState(3);

  const [todos, dispatch] = useReducer(todosReducer, initialTodos);

  function addTodo(title: string): void {
    dispatch({ type: "addTodo", payload: title });
  }

  function changeCompleted(id: number, completed: boolean): void {
    dispatch({
      type: "changeCompleted",
      payload: { id: id, completed: completed },
    });
  }

  function deleteTodo(id: number): void {
    dispatch({ type: "deleteTodo", payload: id });
  }

  async function loadTodosFromAPI() {
    const apiTodos = await fetchTodos();
    dispatch({ type: "loadTodos", payload: apiTodos });
  }
  // useEffect(() => {
  //   loadTodosFromAPI();
  // }, []);

  function saveTodosToLS() {
    localStorage.setItem("todos", JSON.stringify(todos));
  }
  function loadTodosFromLS() {
    const lsTodosString = localStorage.getItem("todos");
    if (lsTodosString) {
      dispatch({ type: "loadTodos", payload: JSON.parse(lsTodosString) });
    }
  }
  useEffect(() => {
    saveTodosToLS();
  }, [todos]);

  return (
    <div>
      <Rating value={rating1} onChange={setRating1} />
      <h1>Todo</h1>
      <NavLink to="/">home</NavLink> <NavLink to="/about">about</NavLink>
      <Routes>
        <Route
          path="/"
          element={
            <div>
              <div>
                <button onClick={loadTodosFromAPI}>load todos from API</button>
                <button onClick={saveTodosToLS}>
                  save todos to local storage
                </button>
                <button onClick={loadTodosFromLS}>
                  load todos from local storage
                </button>
              </div>
              <ul style={{ maxHeight: 400, overflow: "auto" }}>
                {todos.map((todo) => (
                  <TodoItem
                    key={todo.id}
                    title={todo.title}
                    completed={todo.completed}
                    onCompletedChange={(completed) => {
                      changeCompleted(todo.id, completed);
                    }}
                    onDelete={() => deleteTodo(todo.id)}
                  />
                ))}
              </ul>
              <AddTodo onAdd={(newTitle) => addTodo(newTitle)} />
              <AddTodoClass onAdd={(newTitle) => addTodo(newTitle)} />
            </div>
          }
        />
        <Route path="/about" element={<AboutPage />} />
      </Routes>
    </div>
  );
}

export default App;
