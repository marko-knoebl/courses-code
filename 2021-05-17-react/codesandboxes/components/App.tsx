import "./styles.css";
import { useState } from "react";
import ProgressBar from "./ProgressBar";
import Rating from "./Rating";

export default function App() {
  const [prog, setProg] = useState(0.75);
  const [prog2, setProg2] = useState(0.5);
  const [rating, setRating] = useState(3);
  const [rating2, setRating2] = useState(4);

  return (
    <div className="App">
      <h1>Progress bar</h1>
      <ProgressBar value={prog} />
      <ProgressBar value={prog2} color="green" />
      <h1>Rating</h1>
      <Rating value={rating} onChange={(newRating) => setRating(newRating)} />
      <Rating
        value={rating2}
        onChange={(newRating2) => setRating2(newRating2)}
      />
      <div>average rating: {(rating + rating2) / 2}</div>
    </div>
  );
}
