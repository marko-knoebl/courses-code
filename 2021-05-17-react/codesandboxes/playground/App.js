import React, { useState } from "react";
import { FontSizeDemo } from "./FontSizeDemo";

const baseUrl = "https://picsum.photos/300/200?image=";
function SlideshowApp() {
  // Slideshow

  const [img, setImg] = useState(0);

  const imgUrl = `${baseUrl}${img}`;

  const buttonStyle = {
    color: "white",
    backgroundColor: "navy"
  };

  function prevImg() {
    if (img > 0) {
      setImg(img - 1);
    }
  }

  function randImg() {
    setImg(Math.floor(Math.random() * 100));
  }

  // Passwords

  const [password1, setPassword1] = useState("");
  const [password2, setPassword2] = useState("");
  let message;
  if (!(password1.length >= 4)) {
    message = "Pasword 1 is not long enough.";
  } else if (!(password1 === password2)) {
    message = "Passwords don't match.";
  } else {
    message = "";
  }

  // react methods

  const reactMethods = [];
  for (let method in React) {
    reactMethods.push(<li>{method}</li>);
  }

  // todo list

  const initialTodos = [
    { id: 1, title: "groceries", completed: false },
    { id: 2, title: "cooking", completed: true },
    { id: 3, title: "gardening", completed: false }
  ];
  const [todos, setTodos] = useState(initialTodos);

  return (
    <>
      <FontSizeDemo />
      <h1>Image {img}</h1>
      <div style={{ backgroundColor: "lightgrey", display: "flex" }}>
        <button style={buttonStyle} onClick={() => setImg(0)}>
          start
        </button>
        <button style={buttonStyle} onClick={prevImg}>
          prev
        </button>
        <img src={imgUrl} alt="slideshow" style={{ display: "block" }} />
        <button style={buttonStyle} onClick={() => setImg(img + 1)}>
          next
        </button>
        <button style={buttonStyle} onClick={randImg}>
          random
        </button>
      </div>
      <div>
        <label>
          password:
          <input
            value={password1}
            onChange={(event) => setPassword1(event.target.value)}
          />
        </label>
        <label>
          confirm password:
          <input
            value={password2}
            onChange={(event) => setPassword2(event.target.value)}
          />
        </label>
        <div>{message}</div>
        {password1 === password2 ? "Passwords match" : "passwords don't match"}
      </div>
      <h2>React props</h2>
      <ul>{reactMethods}</ul>
      <h2>Todos</h2>
      <ul>
        {todos.map((todo) => (
          <li key={todo.id}>{todo.title}</li>
        ))}
      </ul>
    </>
  );
}
export default SlideshowApp;

// Datenverwaltung ohne Mutationen

// Einzelnes Todo

function withCompletedToggled(todo) {
  return { ...todo, completed: !todo.completed };
}

function withTitleChanged(todo, newTitle) {
  return { ...todo, title: newTitle };
}

const todo1 = { id: 1, title: "foo", completed: false };

const todo2 = withCompletedToggled(todo1);
// { id: 1, title: 'foo', completed: true }

const todo3 = withTitleChanged(todo2, "bar");
// { id: 1, title: 'bar', completed: true}

console.log(todo1);
console.log(todo2);
console.log(todo3);

// Array von Todos

function withNewTodoAdded(todos, newTitle) {
  let maxId = 0;
  for (let todo of todos) {
    if (todo.id > maxId) {
      maxId = todo.id;
    }
  }

  return [...todos, { id: maxId + 1, title: newTitle, completed: false }];
}

function withTodoToggled(todos, id) {
  return todos.map((todo) => {
    if (todo.id === id) {
      // "abändern"
      return { ...todo, completed: !todo.completed };
    } else {
      // nicht "abändern"
      return todo;
    }
  });
}

function withCompletedTodosRemoved(todos) {
  return todos.filter((todo) => !todo.completed);
}

const todos1 = [
  { id: 1, title: "foo", completed: false },
  { id: 2, title: "bar", completed: true }
];

const todos2 = withNewTodoAdded(todos1, "baz");
console.log(todos2);

// schaltet den "completed" - state des todos mit der id 1 um
const todos3 = withTodoToggled(todos2, 1);
console.log(todos3);

const todos4 = withCompletedTodosRemoved(todos3);
console.log(todos4);
