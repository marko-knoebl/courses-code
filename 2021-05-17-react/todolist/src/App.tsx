import { useState } from "react";
import { AppBar, Button } from "@material-ui/core";
import { Theme } from "./types";
import AddTodo from "./AddTodoCls";
import TodoList from "./TodoList";
import AboutView from "./AboutView";
import { NavLink, Route, Routes } from "react-router-dom";
import ThemeContext from "./ThemeContext";
import { useTodos } from "./useTodos";

function App() {

  const {
    todos,
    isLoading,
    addTodo,
    toggleTodo,
    deleteTodo,
    loadTodosAsync,
  } = useTodos()

  // theme

  const [theme, setTheme] = useState<Theme>("light");

  return (
    <ThemeContext.Provider value={theme}>
      <div style={{ backgroundColor: theme === "light" ? "#eee" : "#666" }}>
        <AppBar position="static">
          <h1>Todo</h1>
        </AppBar>

        <nav>
          <NavLink to="/">Home</NavLink> <NavLink to="/about">About</NavLink>
        </nav>

        <Routes>
          <Route
            path="/"
            element={
              <>
                <div>
                  <Button variant="contained" onClick={loadTodosAsync}>
                    load from API
                  </Button>
                  {isLoading ? <div>Loading...</div> : null}
                </div>
                <TodoList
                  todos={todos}
                  onToggle={toggleTodo}
                  onDelete={deleteTodo}
                />
                <AddTodo onAddTodo={addTodo} />
              </>
            }
          />
          <Route path="/about" element={<AboutView />} />
        </Routes>
      </div>
    </ThemeContext.Provider>
  );
}

export default App;
