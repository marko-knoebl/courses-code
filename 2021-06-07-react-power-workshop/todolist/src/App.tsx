import { useState } from "react";
import TodoItem from "./TodoItem";
import AddTodo from "./AddTodo";
import AddTodoClass from "./AddTodoClass";
import { NavLink, Route, Routes } from "react-router-dom";
import AboutView from "./AboutView";
import useTodos from "./useTodos";
import ThemeContext from "./ThemeContext";
import Settings from "./Settings";

function App() {
  const { todos, addTodo, setTodoState, deleteTodo, loadTodos, isLoading } =
    useTodos();

  const [theme, setTheme] = useState<"default" | "dark">("default");

  return (
    <ThemeContext.Provider value={theme}>
      <div>
        <h1>Todo</h1>
        <nav>
          <NavLink to="/">home</NavLink> <NavLink to="/about">about</NavLink>
        </nav>

        <Routes>
          <Route
            path="/"
            element={
              <>
                <Settings theme={theme} setTheme={setTheme} />
                <button
                  onClick={async () => {
                    loadTodos();
                    console.log("fin")
                  }}
                >
                  load todos from API
                </button>
                {isLoading ? <div>loading...</div> : null}
                <ul>
                  {todos.map((todo) => (
                    <TodoItem
                      key={todo.id}
                      title={todo.title}
                      isCompleted={todo.completed}
                      onDelete={() => deleteTodo(todo.id)}
                      onCompletedChange={(completed) =>
                        setTodoState(todo.id, completed)
                      }
                    />
                  ))}
                </ul>
                <AddTodo onAddition={(newTitle) => addTodo(newTitle)} />
                <AddTodoClass onAddition={(newTitle) => addTodo(newTitle)} />
              </>
            }
          />
          <Route path="/about" element={<AboutView />} />
        </Routes>
      </div>
    </ThemeContext.Provider>
  );
}

export default App;
