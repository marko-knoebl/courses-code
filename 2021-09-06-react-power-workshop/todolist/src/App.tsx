import { useContext } from "react";
import { Container, Navbar } from "react-bootstrap";
import { Route, Routes, useNavigate } from "react-router";
import { NavLink } from "react-router-dom";
import AddTodo from "./AddTodo";
import { ThemeContext, useTheme } from "./ThemeContext";
import TodoItem from "./TodoItem";
import useTodos from "./useTodos";

function App() {
  const { todos, addTodo, setTodoStatus, deleteTodo, loadTodos } = useTodos();

  const navigate = useNavigate();

  // const themeObj = useContext(ThemeContext);
  // if (themeObj === undefined) {
  //   throw new Error("no provider for ThemeContext");
  // }
  // const { theme, changeTheme } = themeObj;

  const { theme, changeTheme } = useTheme();

  return (
    <div className="App">
      <Navbar bg={theme} variant={theme === "light" ? "light" : "dark"}>
        <Container>
          <Navbar.Brand>Todo</Navbar.Brand>
        </Container>
      </Navbar>

      <div>
        <button onClick={() => loadTodos()}>load todos from API</button>
      </div>

      <div>
        <button
          onClick={() => changeTheme(theme === "light" ? "dark" : "light")}
        >
          {theme === "light" ? "dark" : "light"}
        </button>
      </div>

      <Container>
        <div>
          <NavLink to="/">Home</NavLink> <NavLink to="/add">Add Todo</NavLink>
        </div>

        <Routes>
          <Route
            path="/add"
            element={
              <AddTodo
                onTodoAdd={(newTitle: string) => {
                  addTodo(newTitle);
                  navigate("/");
                }}
              />
            }
          />

          <Route
            path="/"
            element={
              <ul>
                {todos.map((todo) => (
                  <TodoItem
                    id={todo.id}
                    title={todo.title}
                    completed={todo.completed}
                    onCompletedChange={(completed) =>
                      setTodoStatus(todo.id, completed)
                    }
                    onDelete={() => deleteTodo(todo.id)}
                  />
                ))}
              </ul>
            }
          />
        </Routes>
      </Container>
    </div>
  );
}

export default App;
