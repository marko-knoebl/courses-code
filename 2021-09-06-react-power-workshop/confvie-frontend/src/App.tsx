import React, {Component} from 'react';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
//import jsonData from './config.json';

type Confroom = {
    id: string;
    name: string;
    status: string;
    uri: string;
    callId: string;
}

type AppState = {
    pollingCount: number;
    delay: number;
    confrooms: Array<Confroom>;
    searchField: string;
    expandedRows: Array<string>;
    statusImage: Array<{status: string; img_path: string}>
}

type AppProps = {}

class App extends Component<AppProps, AppState> {

    interval: any;

    constructor(props: {}) {
        super(props);

        this.state = {
            pollingCount: 0,
            delay: 4000,
            confrooms: [],
            searchField: '',
            expandedRows: [],
            statusImage: [
                {
                    status: "free",
                    img_path: ""
                }
            ]
        };
    }

    componentDidMount() {

        this.interval = setInterval(this.tick, this.state.delay) as any;

        /*var config_url = JSON.parse(JSON.stringify(jsonData));
        fetch(config_url["config_url"])
            .then(response => {
                var curr_dyn_config = response.json();
                this.setState({dyn_config: curr_dyn_config});
                return curr_dyn_config;
            })
            .then(function (data) {
*/
        //https://dummy.workbenching.com/confrooms.json
        //http://svadckrh01.bg10.bgfe.local:5000/confrooms.json

        //        fetch("http://confview-mw-service-fb.k8s.bg10.bgfe.local/confrooms")
        //            .then(response => response.json())
        //            .then(users => this.setState({confrooms: users["confrooms"]}));
        //        console.log(this.state.dyn_config);
        
        //            }
        //            .bind(this));

    }

    componentDidUpdate(prevProps: AppProps, prevState: AppState) {
        if (prevState.delay !== this.state.delay) {
            clearInterval(this.interval);
            this.interval = setInterval(this.tick, this.state.delay);
        }
    }

    handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({searchField: e.target.value})
    }

    tick = () => {
        this.setState({
            pollingCount: this.state.pollingCount + 1
        });
        console.log("ENV: " + process.env.REACT_APP_API_URL);
        fetch("example.json")
            .then(response => response.json())
            .then(users => this.setState({confrooms: users}));
        // fetch("http://confview-mw.k8s.bg10.bgfe.local/confrooms.json")
        //     .then(response => response.json())
        //     .then(users => this.setState({confrooms: users}));
    }


    handleRowClick(rowId: string) {
        const currentExpandedRows = this.state.expandedRows;
        const isRowCurrentlyExpanded = currentExpandedRows.includes(rowId);

        const newExpandedRows = isRowCurrentlyExpanded ?
            currentExpandedRows.filter(id => id !== rowId) :
            currentExpandedRows.concat(rowId);

        this.setState({expandedRows: newExpandedRows});
    }

    renderItem(item: Confroom) {
        const clickCallback = () => this.handleRowClick(item.id);
        let status_image;
        if (item.status === "free") {
            status_image = "https://img.icons8.com/emoji/48/000000/green-circle-emoji.png"
        } else {
            status_image = "https://img.icons8.com/emoji/48/000000/red-circle-emoji.png"
        }
        const itemRows = [
            <tr onClick={clickCallback} key={"row-data-" + item.id}>
                {/* if currentrow != colapsed -> arrow down */}
                <td className="thumbnail"><i className="arrow right"/></td>
                <td className="thumbnail"><img alt="Status Thumbnail" src={status_image}/></td>
                <td>{item.name}</td>
                <td>{item.uri}</td>
            </tr>
        ];

        if (this.state.expandedRows.includes(item.id)) {
            itemRows.push(
                <tr key={"row-expanded-" + item.id}>
                    <td colSpan={4}>
                        {item.id}<br/>
                        {item.callId}<br/>
                    </td>
                </tr>
            );
        }

        return itemRows;
    }


    render() {
        let allItemRows: Array<React.ReactNode> = [];
        const {confrooms, searchField} = this.state;
        const filteredconfrooms = confrooms.filter(person =>
            person.name.toLowerCase().includes(searchField.toLowerCase())
        )

        filteredconfrooms.forEach(item => {
            const perItemRows = this.renderItem(item);
            allItemRows = allItemRows.concat(perItemRows);
        });

        return (
            <div className="App">
                <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
                    <h1>Navbar</h1>
                </nav>
                <input className='search form-control'
                       type='search'
                       placeholder='Bitte Namen eingeben'
                       onChange={this.handleChange}
                />
                <table className='table table-stripe table-border'>
                    <thead>
                    <tr>
                        <th></th>
                        <th>Status</th>
                        <th>Name</th>
                        <th>Address</th>
                    </tr>
                    </thead>
                    <tbody>
                    {allItemRows}
                    </tbody>
                </table>
            </div>
        );
    }
}

export default App;
