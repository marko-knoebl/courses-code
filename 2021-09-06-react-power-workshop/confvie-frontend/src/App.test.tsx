import React from 'react';
import { render, screen, within } from '@testing-library/react';
import App from './AppCls';

import exampleData from "../public/example.json";

test('renders a heading with text "Navbar"', () => {
  render(<App />);
  const navbarHeading = screen.getByRole("heading", {name: /Navbar/i});
  expect(navbarHeading).toBeInTheDocument();
});

function double(n: number) {
  return n * 2;
}

test("doubles a number", () => {
  expect(double(3)).toEqual(6);
});

test("renders a table", () => {
  render(<App />);
  const table = screen.getByRole("table");
  expect(table).toBeInTheDocument();
});

test("renders a table header row", () => {
  render(<App />);
  const tableRows = screen.getAllByRole("row");
  const headerRow = tableRows[0];
  const statusHeader = within(headerRow).getByRole(
    "columnheader", {name: /status/i}
  );
  expect(statusHeader).toBeInTheDocument();
});

// test("loads and renders first row", async () => {
//   const data = {
//     value: 'Chuck Norris counted to infinity. Twice.',
//   };
//   globalThis.fetch = () =>
//     Promise.resolve({ json: () => Promise.resolve(exampleData) });
//   render(<App />);
//   const tableBody = screen.getAllByRole("rowgroup")[1];
//   const topRow = await within(tableBody).findByRole("row", {}, {timeout: 3000});
//   expect(topRow).toBeInTheDocument();
// });

test('matches the snapshot', () => {
  render(
    <App />
  );
  const app = screen.getByTestId('app');
  expect(app).toMatchSnapshot();
});
