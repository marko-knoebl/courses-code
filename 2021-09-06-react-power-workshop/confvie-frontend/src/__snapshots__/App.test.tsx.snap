// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  class="App"
  data-testid="app"
>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <h1>
      Navbar
    </h1>
  </nav>
  <input
    class="search form-control"
    placeholder="Name"
    type="search"
  />
  <table
    class="table table-stripe table-border"
  >
    <thead>
      <tr>
        <th />
        <th>
          Status
        </th>
        <th>
          Name
        </th>
        <th>
          Address
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
